File: Data Structures & Algorithms/maximum-subarray/submission-2.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        vector<int> dp(nums);
        int max_sum = dp[0];

        for (int i = 1; i < nums.size(); i++){
            dp[i] = max(nums[i], nums[i] + dp[i-1]);
            if (dp[i] > max_sum) max_sum = dp[i];
        }
        return max_sum;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.py
import bisect
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        sub = []

        for x in nums:
            idx = bisect.bisect_left(sub, x)

            if idx == len(sub):
                sub.append(x)
            else:
                sub[idx] = x
        return len(sub)